fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -145,101 +157,159 @@ private static int getFhirResourceFromBufferReader(BufferedReader resReader, int
+                parseFailures = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), numOfLinesToSkip, fhirResources);
+            } else {
+                parseFailures = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), 0, fhirResources);
             }
+        } catch (Exception ex) {
+            // Clean up.
+            fhirResources.clear();
+            cleanup4TransientUserData(transientUserData, true);
+            // Log the error and throw exception to fail the job, the job can be continued from the current checkpoint after the problem is solved.
+            logger.warning("readFhirResourceFromLocalFile: " + "Error proccesing file " + filePath + " - " + ex.getMessage());
+            throw ex;
         }
-        return exported;
+
+        return parseFailures;
     }
 
 
+    /**
+     * @param dataUrl - URL to the ndjson file.
+     * @param numOfLinesToSkip - number of lines to skip before read.
+     * @param fhirResources - List holds the FHIR resources.
+     * @param transientUserData - transient user data for the chunk.
+     * @return - number of parsing failures.
+     * @throws Exception
+     */
     public static int readFhirResourceFromHttps(String dataUrl, int numOfLinesToSkip, List<Resource> fhirResources,
-            boolean isReuseInput, ImportTransientUserData transientUserData) {
-        int exported = 0;
-        if (isReuseInput) {
+            ImportTransientUserData transientUserData) throws Exception {
+        int parseFailures = 0;
+        int retryTimes = Constants.IMPORT_RETRY_TIMES;
+        do {
             try {
                 if (transientUserData.getBufferReader() == null) {
                     InputStream inputStream = new URL(dataUrl).openConnection().getInputStream();
                     transientUserData.setInputStream(inputStream);
                     BufferedReader resReader = new BufferedReader(new InputStreamReader(inputStream));
                     transientUserData.setBufferReader(resReader);
+                    // Skip the already processed lines after opening the input stream for first read.
+                    parseFailures = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), numOfLinesToSkip, fhirResources);
+                } else {
+                    parseFailures = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), 0, fhirResources);
+                }
+                break;
+            } catch (Exception ex) {
+                // Clean up.
+                fhirResources.clear();
+                cleanup4TransientUserData(transientUserData, true);
+                logger.warning("readFhirResourceFromHttps: " + "Error proccesing file " + dataUrl + " - " + ex.getMessage());
+                if ((retryTimes--) > 0) {
+                    logger.warning("readFhirResourceFromLocalFile: " + "Retry ...");
```

Comment:
```suggestion
                    logger.warning("readFhirResourceFromLocalFile: Retry ...");
```

no need to split string

Reply:
good catch!